src/dotnet/commands/dotnet-new/Program.cs

```diff
@@ -64,11 +65,37 @@ where resourceName.Contains(templateDir)
                             }
                         }
 
-                        archive.ExtractToDirectory(Directory.GetCurrentDirectory());
+                        string projectDirectory = Directory.GetCurrentDirectory();
+
+                        archive.ExtractToDirectory(projectDirectory);
+
+                        string projectJsonFile = Path.Combine(projectDirectory, "project.json");
```

Comment:
Can you extract this new logic into some aptly named methods? Took me a minute to realize what was going on in this new logic.

Perhaps ReplaceProjectJsonTemplateValues and ReplaceFileTemplateNames ?